front/gatsby/gatsby-browser.js

```diff
@@ -0,0 +1,45 @@
+// Copyright 2018-2020 @paritytech/Nomidot authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { ApolloProvider } from '@apollo/react-hooks';
+import { ApolloClient, InMemoryCache } from 'apollo-boost';
+import { split } from 'apollo-link';
+import { HttpLink } from 'apollo-link-http';
+import { WebSocketLink } from 'apollo-link-ws';
+import { getMainDefinition } from 'apollo-utilities';
+import React from 'react';
+
+const httpLink = new HttpLink({
+  uri: process.env.NODE_ENV === 'production' ? 'https://34.76.106.193:4000' : 'http://localhost:4001', // use port forwarding in dev, use external ip in prod
+});
+
+const wsLink = new WebSocketLink({
+  uri: process.env.NODE_ENV === 'production' ? 'wss://34.76.106.193:4000' : 'ws://localhost:4001',
+  options: {
+    reconnect: true
+  }
+});
+
+const link = split(
+  // split based on operation type
+  ({ query }) => {
+    const definition = getMainDefinition(query);
+    return (
+      definition.kind === 'OperationDefinition' &&
+      definition.operation === 'subscription'
+    );
+  },
+  wsLink,
+  httpLink,
+);
+
+const client = new ApolloClient({
+  link,
+  cache: new InMemoryCache()
+})
+
+// eslint-disable-next-line react/prop-types
+export const wrapRootElement = ({element}) => (
+  <ApolloProvider client={client}>{element}</ApolloProvider>
```

Comment:
I would actually be in favor of removing this file, because it's in JavaScript.

So i would:
- create an `apollo.ts`, exporting a React context provider
- in `pages/index.tsx`, do a `import { ThatProvider } from './path/to/apollo.ts'`
- put that provider next to `AccountsContextProvider`

Reply:
yeah i like that better too, I just went with this during the attempts to solve the subscription bug before, seems this is the "gatsby way" of doing things fwir.